test/Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -2485,23 +2519,83 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
+                }
+
+                // After the app is done with the write loop, the connection reset should be logged.
+                // On Linux and macOS, the connection close is still sometimes observed as a FIN despite the LingerState.
+                Assert.True(connectionResetLogged || (!TestPlatformHelper.IsWindows && connectionFinLogged), "Connection reset not logged.");
+            }
+
+            mockKestrelTrace.Verify(t => t.ConnectionStop(It.IsAny<string>()), Times.Once());
+            Assert.True(requestAborted, "RequestAborted token didn't fire.");
+            Assert.False(loggedHigherThanDebug, "Logged message with level higher than debug.");
+        }
+
+
+        [Theory]
+        [MemberData(nameof(ConnectionAdapterData))]
+        public async Task ClientAbortingConnectionImmediatelyIsNotLoggedHigherThanDebug(ListenOptions listenOptions)
+        {
+            var loggedHigherThanDebug = false;
+
+            var mockLogger = new Mock<ILogger>();
+            mockLogger
+                .Setup(logger => logger.IsEnabled(It.IsAny<LogLevel>()))
+                .Returns(true);
+            mockLogger
+                .Setup(logger => logger.Log(It.IsAny<LogLevel>(), It.IsAny<EventId>(), It.IsAny<object>(), It.IsAny<Exception>(), It.IsAny<Func<object, Exception, string>>()))
+                .Callback<LogLevel, EventId, object, Exception, Func<object, Exception, string>>((logLevel, eventId, state, exception, formatter) =>
+                {
+                    if (logLevel > LogLevel.Debug)
                     {
-                        readCount += await connection.Stream.ReadAsync(scratchBuffer, 0, scratchBuffer.Length);
+                        loggedHigherThanDebug = true;
                     }
 
-                    connection.Socket.Shutdown(SocketShutdown.Send);
+                    Logger.Log(logLevel, eventId, state, exception, formatter);
+                });
+
+            var mockLoggerFactory = new Mock<ILoggerFactory>();
+            mockLoggerFactory
+                .Setup(factory => factory.CreateLogger(It.IsAny<string>()))
+                .Returns(Logger);
+            mockLoggerFactory
+                .Setup(factory => factory.CreateLogger(It.IsIn("Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv",
+                                                               "Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets")))
+                .Returns(mockLogger.Object);
+
+            var testContext = new TestServiceContext(mockLoggerFactory.Object);
+
+            // There's not guarantee that the app even gets invoked in this test. The connection reset can be observed
+            // as early as accept.
+            using (var server = new TestServer(context => Task.CompletedTask, testContext, listenOptions))
+            {
+                using (var connection = server.CreateConnection())
+                {
+                    await connection.Send(
```

Comment:
This test is a bit funky already, but what if we run a loop of these? Around 10-100 times?